src/app/fdctl/run/tiles/fd_gui.c

```diff
@@ -268,6 +269,30 @@ privileged_init( fd_topo_t *      topo,
     FD_LOG_ERR(( "identity_key_path not set" ));
 
   ctx->identity_key = fd_keyload_load( tile->gui.identity_key_path, /* pubkey only: */ 1 );
+
+# if FD_HAS_ZSTD
+  /* zstd compress files */
+  for( fd_http_static_file_t * f = STATIC_FILES; f->name; f++ ) {
+    char const * ext = strrchr( f->name, '.' );
+    if( !ext ) continue;
+    if( !strcmp( ext, ".html" ) ||
+        !strcmp( ext, ".css"  ) ||
+        !strcmp( ext, ".js"   ) ||
+        !strcmp( ext, ".svg"  ) ) {}
+    else continue;
+
+    ulong   zstd_bufsz = *f->data_len;
+    uchar * zstd_buf   = malloc( zstd_bufsz );
```

Comment:
@mmcgee-jump I know usage of `malloc` is generally frowned upon. 
But this is one-time initialization code, so I think it gets a pass.

Reply:
Would still be nice to have these files huge page backed

Reply:
I don't like using ZSTD in privileged init here. Can we move this code to `unprivileged_init`? Also you can just do `fd_wksp_alloc( fd_wksp_containing( gui_ctx ), zstd_bufsz )` to get the memory in the GUI gigantic pages. You could also set `loose_sz` to some minimum value to ensure these allocs succeed.

Reply:
done

Reply:
@mmcgee-jump I'm working on the change now. It has about 100 LOC additional code bloat.